Formula/xmrig.rb

```diff
@@ -0,0 +1,21 @@
+class Xmrig < Formula
+  desc "Monero (XMR) CPU miner"
+  homepage "https://github.com/xmrig/xmrig"
+  url "https://github.com/xmrig/xmrig/archive/v2.3.0.tar.gz"
+  sha256 "1b2c876cf48d9e03754c6c71bbf49e7a5fa46c998de968cb22813de1c3a21c26"
+
+  depends_on "cmake" => :build
+  depends_on "libuv"
+
+  def install
+    mkdir "build" do
+      system "cmake", "..", "-DCMAKE_BUILD_TYPE=Release", "-DUV_LIBRARY=#{Formula["libuv"].opt_lib}/libuv.a", *std_cmake_args
+      system "make"
+      bin.install "xmrig"
+    end
+  end
+
+  test do
+    system "#{bin}/xmrig", "-V"
```

Comment:
Can we try to connect to an invalid server name and check an error message?

Reply:
Please note that the program won't die even if an invalid server name is passed, it will just loop in the background hoping it could be resolved one day.

Also, the version update from 2.3.0 to 2.3.1 is required as the version 2.3.0 have a bug which nothing is spilled to stdout/stderr unless it is a terminal, making check the error message impossible.